cli/src/main/kotlin/commands/UploadResultToSw360Command.kt

```diff
@@ -0,0 +1,142 @@
+
+    override fun run() {
+        val (ortResult, duration) = measureTimedValue { ortFile.readValue<OrtResult>() }
+
+        log.perf {
+            "Read ORT result from '${ortFile.name}' (${ortFile.formatSizeInMib}) in ${duration.inMilliseconds}ms."
+        }
+
+        val sw360Config = globalOptionsForSubcommands.config.scanner?.storages?.values
+            ?.filterIsInstance<Sw360StorageConfiguration>()?.singleOrNull()
+
+        requireNotNull(sw360Config) {
+            "No SW360 storage is configured for the scanner."
+        }
+
+        createSw360Connection(sw360Config)?.let { sw360Connection ->
+            val sw360ReleaseClient = sw360Connection.releaseAdapter
+
+            ortResult.getPackages(omitExcluded = true).map { (pkg, _) ->
+                try {
+                    val release = SW360Release()
+                        .setMainLicenseIds(pkg.declaredLicenses)
+                        .setName(pkg.id.name)
+                        .setVersion(pkg.id.version)
+
+                    sw360ReleaseClient.createRelease(release)?.let {
+                        log.debug { "Release ${release.name}-${release.version} created in SW360." }
+                    }
+                } catch (e: SW360ClientException) {
+                    log.error { "Could not create the release in SW360 for the reason: ${e.message}" }
+                }
+            }
+        }
+    }
+
+    /**
+     * Crates a [SW360Connection] based on the given [Sw360StorageConfiguration].
+     */
+    private fun createSw360Connection(config: Sw360StorageConfiguration): SW360Connection? {
+        require(config.restUrl.isNotBlank()) {
+            "URL for SW360 REST API is missing."
+        }
+
+        require(config.authUrl.isNotBlank()) {
+            "URL for SW360 authentication server is missing."
+        }
+
+        require(config.username.isNotBlank()) {
+            "Username for SW360 is missing."
+        }
+
+        require(config.password.isNotBlank()) {
+            "Password of the username for SW360 is missing."
+        }
+
+        require(config.clientId.isNotBlank()) {
+            "Client ID of the SW360 instance is missing."
+        }
+
+        require(config.clientPassword.isNotBlank()) {
+            "Client password of the SW360 instance is missing."
+        }
+
+        return SW360ConnectionFactory()
+            .newConnection(SW360ClientConfig.createConfig(
+                config.restUrl,
+                config.authUrl,
+                config.username,
+                config.password,
+                config.clientId,
+                config.clientPassword,
+                ServiceFactory().createHttpClient(
+                    config.proxyHost.isNullOrEmpty() && config.proxyPort!! > 0,
+                    config.proxyHost,
+                    config.proxyPort!!
+                ),
+                ServiceFactory.getObjectMapper()
+            ))
```

Comment:
When the `require()` checks are moved and this becomes the only call in the function, I'd probably remove this `createSw360Connection()` function and inline its contents into the caller.